CONTRIBUTING.md

```diff
@@ -0,0 +1,209 @@
+
+## Getting Started
+
+Requirements:
+
+- Node 14+
+- Yarn
+- Docker
+  - Used in the tests. Docker daemon must be running.
+- Localstack
+  - Used in the tests.
+- [AWS-CDK](https://docs.aws.amazon.com/cdk/v2/guide/cli.html) - 2.x
+
+Fork the repo and then run:
+
+```console
+$ git clone https://github.com/{your-account}/functionless.git
+$ cd functionless
+$ yarn install
+```
+
+We do all of our work in [`VSCode`](https://code.visualstudio.com/) and include configuration files to improve the developer experience. There is no strict dependency. Suggested plugins: vscode-jest, prettier-vscode, streetsidesoftware.code-spell-checker.
+
+### Layout
+
+- `/` and `/src` - The Functionless Package - containing compiler (`src/compiler.ts`), transpilers (`src/vtl.ts`), and publicly exposed resources (`Function`, `StepFunction`, etc). Generates the [functionless](https://www.npmjs.com/package/functionless) NPM repo.
+- `/test` - Tests for the main repo, see [testing](#testing)
+- `/website` - Functionless.org code
+- `/test-app` - A series of test CDK applications which consume `functionless` via npm. We use this to test compilation of various "real world" scenarios and CDK synthesis.
+
+Functionless leverages the [typescript compiler api](https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API). `compile.ts` runs over the ts of a consuming application, validating, transforming, and extracting necessary information.
+
+> Note: Functionless does not compile itself, at the moment, functionless like `reflect` cannot be used to transform the core functionless source. `/test`s are compiled.
+
+The compiler outputs the functionless AST (`/src/expression.ts`, `/src/declaration.ts`, `/src/statement.ts`) into the transpiled javascript which is used by the resources (`/src/function.ts`, `/src/step-function.ts`, etc) during **cdk synthesis** to infer infrastructure (CDK Constructs) and transpile typescript/Functionless AST to cloud native languages (ASL, Velocity).
+
+## Building Functionless
+
+```console
+$ yarn install
+$ yarn watch
+```
+
+### Guidelines
+
+- Every change or bug fix MUST come with one or more tests that would break before the change.
+- Errors MUST be documented as [Error Codes](#error-codes) with detailed descriptions.
+- Errors SHOULD be discovered via `/src/validate.ts`. `/src/validate.ts` runs with the `functionless` cli and the `@functionless/language-service` in the IDE to give early errors to consumers.
+- No/Minimal Magic - Core Functionless resources should improve the developer experience without introducing opinionated implementation. The goal is to translate typescript behavior into AWS services. The infrastructure created should be clear and logical. For example, `Step Functions` cannot perform arithmetic natively, we will not automatically create/call a lambda to do it. We will validate for arithmetic, throw an error, and show how a `Function` can be used as a workaround.
```

Comment:
We may introduce higher level constructs that do poly fill, but that behavior will be explicitly chosen by the user